File: DependencyInjection/DependencyInjection/_1_ControlFreak/Inbound/BinaryParsing.cs
using DependencyInjection._1_ControlFreak.InMessages;
using DependencyInjection._1_ControlFreak.Interfaces;

namespace DependencyInjection._1_ControlFreak.Inbound
{
  class BinaryParsing
  {
    public AcmeMessage ResultFor(byte[] frameData)
    {
      if (frameData == null)
      {
        return new NullMessage();
      }
      else if (frameData[0] == 1)
      {
        return new StartMessage();
      }
      else
      {
        return new StopMessage();
      }
      
    }
  }
}

File: Java/dependency-injection/control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Inbound/BinaryParsing.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Inbound;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.InMessages.NullMessage;
import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.InMessages.StartMessage;
import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.InMessages.StopMessage;
import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Interfaces.AcmeMessage;

class BinaryParsing {
  public AcmeMessage resultFor(byte[] frameData) {
    if (frameData == null) {
      return new NullMessage();
    } else if (frameData[0] == 1) {
      return new StartMessage();
    } else {
      return new StopMessage();
    }

  }
}
